framework/db/CDbTransaction.php

```diff
@@ -61,7 +61,8 @@ public function commit()
 		if($this->_active && $this->_connection->getActive())
 		{
 			Yii::trace('Committing transaction','system.db.CDbTransaction');
-			$this->_connection->getPdoInstance()->commit();
+			if($this->_connection->getPdoInstance()->inTransaction())
```

Comment:
@samdark Shouldn't the trace message be part of the if-block?

Reply:
Yes, I think so.